[wip]  Conda build (#318)
man-group/ArcticDB: cpp/arcticdb/CMakeLists.txt

option(HIDE_LINKED_SYMBOLS "Passed `exclude-libs` through to linker, ensuring linked library symbols are not exported. Shrinks binary size." ON)
option(STATIC_LINK_STD_LIB "Statically link the C++ standard libraries" ON)

> 

Reviewer: ```suggestion
```

find_package(LZ4 REQUIRED)
find_package(prometheus-cpp CONFIG REQUIRED)
find_package(Threads REQUIRED)
if(NOT ARCTICDB_USING_CONDA)
    find_package(unofficial-pcre REQUIRED)
else()
    find_package(PCRE REQUIRED)

> endif()

Reviewer: Can this be moved bellow?

Author: sure! not in love with any change I made, whatever makes is nice is fine

        fmt::fmt
        ${Protobuf_lite_LIBRARY}
        ${Protobuf_LIBRARY}
        ${LZ4_LIBRARY}
        ${Zstd_LIBRARY}
        lmdb
        ${ARCTICDB_MONGO_LIBS}
        spdlog::spdlog
        Folly::folly # Transitively includes: double-conversion, gflags, glog, libevent, libssl, libcrypto, libiberty, libsodium
        remotery_static
        )

> # PCRE 
> if(${ARCTICDB_USING_CONDA})
>     list (APPEND arcticdb_core_libraries pcre)
> else()
>     list (APPEND arcticdb_core_libraries unofficial::pcre::pcre)
> endif()
> 
> # no static libraries when using conda
> if(${ARCTICDB_USING_CONDA})
>     add_compile_definitions(ARCTICDB_USING_CONDA)
>     list (APPEND arcticdb_core_libraries xxHash::xxHash)
> else()
>     add_compile_definitions(ARCTICDB_USING_STATIC_XXHASH)
>     list (APPEND arcticdb_core_libraries xxhash_static)
> endif()

Reviewer: I think those can be merged into a single block.

For instance:

```suggestion
# Dependencies expose different names of their libraries depending
# on their origin (i.e. vendored source or packages on conda-forge). 
if(${ARCTICDB_USING_CONDA})
    # ARCTICDB_USING_CONDA is used in header of the files including vendored
    # sources to be able to use dependencies via conda at compile time.
    add_compile_definitions(ARCTICDB_USING_CONDA)
    
    # We must use dynamic libraries.
    list (APPEND arcticdb_core_libraries
        xxHash::xxHash
        pcre
    )
else()

    add_compile_definitions(ARCTICDB_USING_STATIC_XXHASH)
    list (APPEND arcticdb_core_libraries
        xxhash_static
        unofficial::pcre::pcre
    )
endif()
```

        ${AWSSDK_LINK_LIBRARIES}
        )

> #GENERATE_EXPORT_HEADER(arcticdb_core)

Author: this has been some debugging leftover

> #GENERATE_EXPORT_HEADER(arcticdb_core)

Reviewer: Then, this might be reintroduced.
```suggestion
GENERATE_EXPORT_HEADER(arcticdb_core)
```
